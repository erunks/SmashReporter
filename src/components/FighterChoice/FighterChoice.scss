@import '../../styles/base.scss';
$temp: white;

.FighterChoice {
  align-items: center;
  border: 4px solid $temp;
  display: flex;
  filter: drop-shadow(15px 15px 4px $temp);
  justify-content: center;
  margin: auto;
  position: relative;
  transform: skewY(-5deg);
  width: 100%;

  &.faceLeft {
    filter: drop-shadow(-15px 15px 4px $temp);
    transform: skewY(5deg);

    img {
      transform: scale(-1,1);
    }
  }

  &:hover {
    .helpText {
      @include visible(flex);
    }
  }

  .helpText {
    @include absolute-position;
    @include hidden;

    align-items: center;
    background-color: rgba(1, 1, 1, 0.5);
    color: white;
    font-variant: small-caps;
    height: 100%;
    margin: auto 0;
    text-align: center;
    width: 100%;
  }
}