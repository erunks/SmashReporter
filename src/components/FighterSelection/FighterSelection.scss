@import '../../styles/base.scss';

.FighterSelection {
  align-items: center;
  display: flex;
  justify-content: center;
  flex-direction: column;

  .fighterRow:first-child {
    border-top: 2px solid black;
  }
  
  @include breakpoint($sm) {
    grid-column: span 2/2;
    grid-row: 1;
  }

  @include breakpoint($xl) {
    grid-column: 2;
  }
}
