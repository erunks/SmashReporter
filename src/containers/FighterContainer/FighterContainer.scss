@import '../../styles/base.scss';

.FighterContainer {
  align-items: center;
  display: grid;
  grid-auto-flow: row dense;
  justify-content: center;

  @include breakpoint($sm) {
    grid-template-columns: auto auto;
  }

  @include breakpoint($xl) {
    grid-template-columns: 15% auto 15%;
  }
}
