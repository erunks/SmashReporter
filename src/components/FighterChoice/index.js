import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import { buttonize } from 'helpers/jsx-a11y-helper'
import { URLS } from 'lib/imageAssets';

import './FighterChoice.scss';

const FighterChoice = ({ leftFacing, fighter, onClick }) => (
    <Fragment>
      <div
        className={`FighterChoice vertical ${leftFacing ? 'faceLeft' : ''}` }
        style={{ backgroundImage: `url(${URLS.fighterThumbnail(fighter)})` }}
        alt={`fighter: ${fighter}`} { ...buttonize(onClick) }
      >
        <div className="helpText center-container">
          <p>Click to Select Fighter</p>
        </div>
      </div>
      <div
        className={`FighterChoice horizontal ${leftFacing ? 'faceLeft' : ''}`}
        style={{ backgroundImage: `url(${URLS.fighterThumbnailIcon(fighter)})` }}
        alt={`fighter: ${fighter}`} {...buttonize(onClick)}
      >
        <div className="helpText center-container">
          <p>Click to Select Fighter</p>
        </div>
      </div>
    </Fragment>
);

const {
  bool,
  func,
  string
} = propTypes;

FighterChoice.propTypes = {
  leftFacing: bool,
  fighter: string.isRequired,
  onClick: func.isRequired
};

export default FighterChoice;
