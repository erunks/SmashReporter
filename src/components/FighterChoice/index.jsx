import React from 'react';
import propTypes from 'prop-types';
import { buttonize } from '../../helpers/jsx-a11y-helper'
import { URLS } from '../../lib/imageAssets';

import './FighterChoice.scss';

const FighterChoice = ({ leftFacing, fighter, onClick }) => (
    <div className={`FighterChoice ${leftFacing ? 'faceLeft' : ''}` } { ...buttonize(onClick) }>
    <img src={URLS.fighterThumbnail(fighter)} alt={`fighter: ${fighter}`} />
      <div className="helpText">
        <p>Click to Select Fighter</p>
      </div>
    </div>
);

const {
  bool,
  func,
  string
} = propTypes;

FighterChoice.propTypes = {
  leftFacing: bool,
  fighter: string.isRequired,
  onClick: func.isRequired
};

export default FighterChoice;
