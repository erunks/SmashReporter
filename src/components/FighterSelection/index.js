import React from 'react';
import propTypes from 'prop-types';
import { useRecoilValue } from 'recoil';
import { orderedFighters } from 'recoils/fighter';
import FighterRow from 'components/FighterRow';
import chunk from 'lodash/chunk';

import './FighterSelection.scss';

const FighterSelection = () => {
  const fighters = useRecoilValue(orderedFighters);

  return (
    <div className="FighterSelection">
      {
        chunk(fighters, 12).map(
          (fighterChunk, index) => <FighterRow
              key={`fighterRow${index}`}
              fighterChunk={fighterChunk}
            />
        )
      }
    </div>
  )
};

const {
  arrayOf,
  object
} = propTypes;

FighterSelection.propTypes = {
  initFighters: arrayOf(object)
};

export default FighterSelection;
