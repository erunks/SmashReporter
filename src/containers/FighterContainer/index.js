import React, { useEffect } from 'react';
import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
import { fighterState, selectedFightersState } from 'recoils/fighter';
import { selectingPlayerState } from 'recoils/player';
import FighterChoice from 'components/FighterChoice';
import FighterSelection from 'components/FighterSelection';
import MainLayout from 'containers/MainLayout';
import { getFighters } from 'lib/api';

import './FighterContainer.scss';

const FighterContainer = () => {
  const selectedFighters = useRecoilValue(selectedFightersState);
  const [fighters, setFighters] = useRecoilState(fighterState);
  const setSelectingPlayer = useSetRecoilState(selectingPlayerState);

  const fightersPopulated = () => Object.getOwnPropertyNames(fighters[0]).length

  useEffect(() => {
    getFighters().then((fighters) => {
      setFighters(fighters);
    })
  }, [setFighters]);

  return (
    <MainLayout>
      <div className="center-container">
        { fightersPopulated() && <div className="FighterContainer">
          <FighterChoice
            leftFacing={true}
            fighter={selectedFighters[0]}
            onClick={() => setSelectingPlayer(0)}
          />
          <FighterSelection/>
          <FighterChoice
            fighter={selectedFighters[1]}
            onClick={() => setSelectingPlayer(1)}
          />
        </div>}
      </div>
    </MainLayout>
  )
};

export default FighterContainer;
