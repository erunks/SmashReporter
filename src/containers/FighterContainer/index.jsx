import React, { useState } from 'react';
import FighterChoice from '../../components/FighterChoice/index.jsx';
import FighterSelection from '../../components/FighterSelection/index.jsx';
import './FighterContainer.scss';

const FighterContainer = () => {
  const [playerOneFighter, setplayerOneFighter] = useState('mario');
  const [playerTwoFighter, setplayerTwoFighter] = useState('mario');
  const [selectingPlayer, setSelectingPlayer] = useState(0);


  const setPlayerFighter = (name) => {
    if (selectingPlayer === 0) {
      setplayerOneFighter(name);
    } else if (selectingPlayer === 1) {
      setplayerTwoFighter(name);
    }
  };

  return (
    <div className="FighterContainer">
      <FighterChoice
        leftFacing={true}
        fighter={playerOneFighter}
        onClick={() => setSelectingPlayer(0)}
      />
      <FighterSelection
        setPlayerFighter={setPlayerFighter}
        selectingPlayer={selectingPlayer}
      />
      <FighterChoice
        fighter={playerTwoFighter}
        onClick={() => setSelectingPlayer(1)}
      />
    </div>
  )
};

export default FighterContainer;
