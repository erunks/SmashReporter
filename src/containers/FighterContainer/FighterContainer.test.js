import React from 'react';
import { fireEvent, mountWrapper, wrappedRender } from 'helpers/test-helper';
import { fighterState } from 'recoils/fighter';
import { selectingPlayerState } from 'recoils/player';
import FighterChoice from 'components/FighterChoice';
import FighterSelection from 'components/FighterSelection';
import FighterContainer from '.';

describe('<FighterContainer>', () => {
  describe('when the fighters are not yet populated', () => {
    const container = mountWrapper(
      <FighterContainer />,
      { initializeState: (snap) => snap.set(fighterState, [{}]) }
    );

    it('should match the snapshot', () => {
      expect(container.html()).toMatchSnapshot();
    });

    it('should not render its subcomponents', () => {
      expect(container.find(FighterChoice).length).toEqual(0);
      expect(container.find(FighterSelection).length).toEqual(0);
    });
  });

  describe('when the fighters are populated', () => {
    const fighterStateData = [{
      dlc: 0,
      id: 1,
      name: 'Mario'
    }];

    const container = mountWrapper(
      <FighterContainer />,
      { initializeState: (snap) => snap.set(fighterState, fighterStateData) }
    );

    it('should match the snapshot', () => {
      expect(container.html()).toMatchSnapshot();
    });

    it('should contain all the appropriate subcomponents', () => {
      expect(container.find(FighterChoice).length).toEqual(2);
      expect(container.find(FighterSelection).length).toEqual(1);
    });
  });

  describe('functionality', () => {
    const fighterStateData = [
      {
        dlc: 0,
        id: 1,
        name: 'Mario'
      },
      {
        dlc: 0,
        id: 99,
        name: 'Random'
      },
    ];

    test('when the fighter choice is clicked, the selection should update accordingly', () => {
      const { unmount } = wrappedRender(
        <FighterContainer />,
        {
          initializeState: (snap) => {
            snap.set(fighterState, fighterStateData);
            snap.set(selectingPlayerState, 0);
          }
        }
      );

      const playerOneSelection = document.querySelector('.selection.player-one');
      expect(playerOneSelection).toBeInTheDocument();
      const fighterChoices = document.querySelectorAll('.FighterChoice');
      expect(fighterChoices.length).toBe(4);

      const playerTwoFighterChoice = fighterChoices[3];
      fireEvent.click(playerTwoFighterChoice);
      const playerTwoSelection = document.querySelector('.selection.player-two');
      expect(playerTwoSelection).toBeInTheDocument();

      unmount();
    });

    test('when a fighter is chosen, the fighter choice should update', () => {
      const { unmount } = wrappedRender(
        <FighterContainer />,
        {
          initializeState: (snap) => {
            snap.set(fighterState, fighterStateData);
            snap.set(selectingPlayerState, 0);
          }
        }
      );

      const playerOneFighterChoice = document.querySelectorAll('.FighterChoice')[0];
      expect(playerOneFighterChoice).toBeInTheDocument();
      expect(playerOneFighterChoice.getAttribute('alt')).toBe('fighter: random');

      const fighter = document.querySelector('.fighter');
      fireEvent.click(fighter);
      expect(playerOneFighterChoice.getAttribute('alt')).toBe('fighter: mario');

      unmount();
    });
  });
});
