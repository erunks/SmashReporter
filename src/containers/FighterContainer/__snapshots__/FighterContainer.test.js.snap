// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FighterContainer> when the fighters are not yet populated should match the snapshot 1`] = `"<div class=\\"MainLayout\\"><header class=\\"Header\\"><h1>Smash Reporter</h1></header><main><div class=\\"center-container\\">0</div></main><footer class=\\"Footer\\"><span>Edward Runkel<i>©2020</i></span></footer></div>"`;

exports[`<FighterContainer> when the fighters are populated should match the snapshot 1`] = `"<div class=\\"MainLayout\\"><header class=\\"Header\\"><h1>Smash Reporter</h1></header><main><div class=\\"center-container\\"><div class=\\"FighterContainer\\"><div class=\\"FighterChoice vertical faceLeft\\" style=\\"background-image: url(https://ik.imagekit.io/ozcerk4wii/fighter/thumb_v/random.png);\\" alt=\\"fighter: random\\" role=\\"button\\"><div class=\\"helpText center-container\\"><p>Click to Select Fighter</p></div></div><div class=\\"FighterChoice horizontal faceLeft\\" style=\\"background-image: url(https://ik.imagekit.io/ozcerk4wii/fighter/thumb_a/random.png);\\" alt=\\"fighter: random\\" role=\\"button\\"><div class=\\"helpText center-container\\"><p>Click to Select Fighter</p></div></div><div class=\\"FighterSelection\\"><div class=\\"fighterRow\\"><div class=\\"fighter\\" role=\\"button\\"><div class=\\"fighterCard\\" alt=\\"fighter: Mario\\"><img alt=\\"fighter: Mario\\" src=\\"https://ik.imagekit.io/ozcerk4wii/fighter/thumb_a/mario.png?ik-sdk-version=react-1.0.8&amp;tr=q-20%2Cbl-6\\"><div class=\\"selection player-one\\"><div class=\\"topLeft\\"></div><div class=\\"bottomRight\\"></div></div></div></div></div></div><div class=\\"FighterChoice vertical \\" style=\\"background-image: url(https://ik.imagekit.io/ozcerk4wii/fighter/thumb_v/random.png);\\" alt=\\"fighter: random\\" role=\\"button\\"><div class=\\"helpText center-container\\"><p>Click to Select Fighter</p></div></div><div class=\\"FighterChoice horizontal \\" style=\\"background-image: url(https://ik.imagekit.io/ozcerk4wii/fighter/thumb_a/random.png);\\" alt=\\"fighter: random\\" role=\\"button\\"><div class=\\"helpText center-container\\"><p>Click to Select Fighter</p></div></div></div></div></main><footer class=\\"Footer\\"><span>Edward Runkel<i>©2020</i></span></footer></div>"`;
